public/app/features/correlations/types.ts

```diff
@@ -26,18 +26,41 @@ export interface RemoveCorrelationResponse {
   message: string;
 }
 
-export type CorrelationType = 'query';
+type CorrelationTypeOptions = {
+  value: string;
+  label: string;
+  description: string;
+};
+
+export const CORR_TYPES: Record<string, CorrelationTypeOptions> = {
+  query: {
+    value: 'query',
+    label: 'Query',
+    description: 'Open a query',
+  },
+  external: {
+    value: 'external',
+    label: 'External',
+    description: 'Open an external URL',
+  },
+};
+
+const corrTypeArray = Object.values(CORR_TYPES).map((ct) => ct.value);
+
+export type CorrelationType = (typeof corrTypeArray)[number];
+
+export type ExternalTypeTarget = { url: string };
 
 export interface CorrelationConfig {
   field: string;
-  target: object; // this contains anything that would go in the query editor, so any extension off DataQuery a datasource would have, and needs to be generic
+  target: object | ExternalTypeTarget; // for queries, this contains anything that would go in the query editor, so any extension off DataQuery a datasource would have, and needs to be generic. For external, it simply contains a URL
   transformations?: DataLinkTransformationConfig[];
 }
 
 export interface Correlation {
   uid: string;
   sourceUID: string;
-  targetUID: string;
+  targetUID?: string;
```

Comment:
Wouldn't it make sense to make this a discriminate union? like:
```
Correlation = {
...
targetUID: string,
type: 'query'
} | {
...
// no target UID here
type: 'external'
}
```

Also the `CorrelationConfig` then can be 2 separate types. I think that would be more type safe as the only check needed is the `correlation.type` and based on that TS should be able to figure out the right props for it.

Reply:
All of the weirdness with the typing here is due to something I couldn't figure out something simpler for with the react hook form. Maybe tomorrow if you have time we can screenshare and we cna figure something out with the types. The outcome here I realize is all over the place, but I had issues finding a simpler solution that would work with all the places we need this typing.